faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -82,22 +188,143 @@ object FaunaClient {
+    *
+    * Alternatively, an instance of any class can be passed as parameter as long as there is
+    * an implicit [[faunadb.values.Encoder Encoder]] for it in scope.
+    *
+    * Example:
+    *
+    * {{{
+    * import faunadb.values._
+    * import scala.concurrent.ExecutionContext.Implicits.global
+    *
+    * val result: Future[Value] =
+    *   client.query(
+    *     "Hello, FaunaDB!"
+    *   )
+    * }}}
+    *
+    * ==Issue multiple queries as a single transaction==
+    *
+    * Pass an [[scala.collection.Iterable Iterable]] implementation of [[faunadb.query.Expr Expr]] as parameter.
+    *
+    * Example:
+    *
+    * {{{
+    * import faunadb.values._
+    * import faunadb.query._
+    * import scala.concurrent.ExecutionContext.Implicits.global
+    *
+    * val result: Future[IndexedSeq[Value]] =
+    *   client.query(
+    *     Seq(
+    *       Get(Ref(Collection("users"), "263608813107544596")),
+    *       Get(Ref(Collection("users"), "263608827286389268"))
+    *     )
+    *   )
+    * }}}
+    *
+    * Alternatively, an [[scala.collection.Iterable Iterable]] implementation of any class can be passed as parameter,
+    * as long as there is an implicit [[faunadb.values.Encoder Encoder]] for it in scope.
+    *
+    * Example:
+    *
+    * {{{
+    * import faunadb.values._
+    * import scala.concurrent.ExecutionContext.Implicits.global
+    *
+    * val result: Future[IndexedSeq[Value]] =
+    *   client.query(
+    *     Seq(
+    *       "Hello",
+    *      "FaunaDB!"
+    *     )
+    *   )
+    * }}}
+    *
+    * =Other options=
+    *
+    * ==Set query timeout==
+    *
+    * The query method also accepts an ''implicit'' timeout parameter
+    * of [[scala.concurrent.duration.FiniteDuration FiniteDuration]] type.
+    *
+    * Example:
+    *
+    * {{{
+    * import faunadb.query._
+    * import faunadb.values._
+    *
+    * import scala.concurrent.ExecutionContext.Implicits.global
+    * import scala.concurrent.duration._
+    *
+    * implicit val timeout: FiniteDuration = 500 millis
+    *
+    * val result: Future[Value] =
+    *   client.query(
+    *     Get(Ref(Collection("users"), "263608813107544596"))
+    *   )
+    * }}}
+    *
+    * The timeout value defines the maximum time a query will be allowed
+    * to run on the server. If the value is exceeded, the query is aborted.
+    *
+    * If a timeout value is provided when calling this method, it replaces
+    * the timeout value set for this [[faunadb.FaunaClient]] (if any) for
+    * the scope of this query.
+    *
+    * The timeout value has milliseconds precision.
+    *
+    * @see [[http://spray.io/blog/2012-12-13-the-magnet-pattern/ Magnet Pattern]]
+    * @see [[faunadb.FaunaClient.QueryAPI.QueryMagnet QueryMagnet]]
+    * */
+  def query(magnet: QueryMagnet): magnet.Result = magnet(json, performRequest _)
+
+  private def performRequest(body: JsonNode, timeout: Option[FiniteDuration], ec: ExecutionContext): Future[Value] = {
```

Comment:
`ExecutionContext` needs to be defined explicitly here because we're lifting the _method_ into a _function_ when passing `performRequest` as a param to the `QueryMagnet.apply()` method. Methods with implicit params cannot be lifted into functions.